proxy/http/HttpTransact.cc

```diff
@@ -7845,10 +7845,12 @@ HttpTransact::build_request(State *s, HTTPHdr *base_request, HTTPHdr *outgoing_r
       DebugTxn("http_trans", "[build_request] adding target to URL for parent proxy");
       outgoing_request->set_url_target_from_host_field();
     }
-  } else if (s->next_hop_scheme == URL_WKSIDX_HTTP || s->next_hop_scheme == URL_WKSIDX_HTTPS ||
-             s->next_hop_scheme == URL_WKSIDX_WS || s->next_hop_scheme == URL_WKSIDX_WSS) {
+  } else if (!(s->client_info.is_transparent && base_request->is_target_in_url()) &&
```

Comment:
It's not equivalent -- the additional () forces both terms to be evaluated (even if the first one is false)